lib/ansible/modules/cloud/amazon/ec2_elb_facts.py

```diff
@@ -201,20 +203,28 @@ def _get_elb_info(self, elb):
 
 
     def list_elbs(self):
-        elb_array = []
-
-        try:
-            all_elbs = self.connection.get_all_load_balancers()
-        except BotoServerError as err:
-            self.module.fail_json(msg = "%s: %s" % (err.error_code, err.error_message))
+        elb_array, token = [], None
 
-        if all_elbs:
-            if self.names:
-                for existing_lb in all_elbs:
-                    if existing_lb.name in self.names:
-                        elb_array.append(existing_lb)
+        while True:
+            try:
+                all_elbs = self.connection.get_all_load_balancers(marker=token)
+                token = all_elbs.next_token
+            except BotoServerError as err:
+                self.module.fail_json(msg = "%s: %s" % (err.error_code, err.error_message),
+                    exception=traceback.format_exc())
+
+            if all_elbs:
+                if self.names:
```

Comment:
As long as `self.names` defaults to `None` AND all names passed in `self.names` exist in the AWS account.

We could optimistically try with `self.names` and if that has an exception fall back to getting all of them. I don't know if the extra conditionals are worth it for that though.